.github/workflows/release.yaml

```diff
@@ -9,48 +8,48 @@ on:
 jobs:
   Release:
     name: CBMC viewer release
-    runs-on: ubuntu-18.04
-
+    runs-on: ubuntu-latest
+    env:
+      GITHUB_TOKEN: ${{ secrets.RELEASE_CI_ACCESS_TOKEN }}
     steps:
-
-      - name: Checkout the repository
+      - name: Checkout code
         uses: actions/checkout@v2
-
-      - name: Set the package version
+      - name: Get version
         run: |
           # The environment variable GITHUB_REF is refs/tags/viewer-*
-          echo "TAG=${GITHUB_REF:10}" >> $GITHUB_ENV
-          echo "VERSION=${GITHUB_REF:17}" >> $GITHUB_ENV
-          printenv | sort
-          cat ${GITHUB_EVENT_PATH}
-
-      - name: Create the package
-        run: |
-          # Patch the version number the source to be consistent with the tag
-          sed -i.bak \
-            "s/NUMBER *=.*/NUMBER = \"${VERSION}\"/" \
-            src/cbmc_viewer/version.py
-
-          # Patch the version number pip setup to be consistent with the tag
-          sed -i.bak \
-            "s/version *=.*/version = ${VERSION}/" \
-            setup.cfg
-
-          # Create the package
-          python3 -m pip install --upgrade pip build setuptools wheel
-          make pip
-
-          # Record the package name
-          # The source *.zip and binary *.whl packages are in dist
-          echo "PACKAGE=$(ls dist/*.whl)" >> $GITHUB_ENV
-
-      - name: Create the release
-        env:
-          # GitHub creates this secret for authentication in a workflow
-          GITHUB_TOKEN: ${{ secrets.RELEASE_CI_ACCESS_TOKEN }}
+          echo "SETUP_VERSION=$(python3 -c "import configparser; config = configparser.ConfigParser(); config.read('setup.cfg'); print(config['metadata']['version'])")" >> $GITHUB_ENV
+          echo "SOURCE_VERSION=$(python3 -c "import src.cbmc_viewer.version; print(src.cbmc_viewer.version.NUMBER)")" >> $GITHUB_ENV
+          echo "TAG_VERSION=$(echo ${{ github.ref }} | cut -d "/" -f 3 | cut -d "-" -f 2)" >> $GITHUB_ENV
+      - name: Version Check
         run: |
-          python3 -m pip install --upgrade pygithub
-          python3 .github/workflows/create_tagged_release.py --verbose \
-            --binary-package ${PACKAGE} \
-            --tag ${TAG} \
-            --version ${VERSION}
+          if [[ ${{ env.SETUP_VERSION }} != ${{ env.TAG_VERSION }} ]] && [[ ${{ env.SOURCE_VERSION }} != ${{ env.TAG_VERSION }} ]]; then
```

Comment:
Shouldn't this be || and not &&?  Does this exit without any error message printed to the log?  Could we add the line 
```
echo Setup and source versions ${{env.SETUP_VERSION}} and ${{env.SOURCE_VERSION}} did not match tag version ${{env.TAG_VERSION}}
```

Reply:
It need to be `&&` right?
In case version is same in in `tag` and `setup.cfg` but not same in `tag` and `version.py`.
With `||` it'll just pass if even one of them is same as `tag` version

Reply:
You want 
```
if not( setup==tag  and  source==tag) then alarm
```
That is equivalent to 
```
if (setup != tag  or source != tag) then alarm
```

Reply:
Resolved